pages/api/api-mocker/[...slug].ts

```diff
@@ -0,0 +1,199 @@
+    );
+  });
+  if (matchPenultimate.length === 1) return matchPenultimate;
+
+  throw new Error(
+    `Failed to match slug: "${slug}" to single action. Candidate paths: ${matches.map(
+      (match) => match.path
+    )}`
+  );
+}
+
+export function getMatchingAction(
+  org_id: number,
+  actions: Action[],
+  method: RequestMethods,
+  slug: string[]
+): Action | null {
+  const slugString = "/" + slug.join("/");
+
+  let matches = actions.filter((action) => {
+    return action.path && slugMatchesPath(action.path, slugString);
+  });
+
+  if (matches.length === 0) {
+    console.log(
+      `Could not match slug "${slugString}" with method "${method}" to any action in the database for organisation "${org_id}"`
+    );
+    return null;
+  }
+
+  if (matches.length > 1) matches = processMultipleMatches(matches, slug);
+
+  console.log(
+    `Successfully matched slug "${slugString}" to action with path "${matches[0].path}"`
+  );
+
+  return matches[0];
+}
+
+export function getPathParameters(
+  path: string,
+  querySlugs: string[]
+): { [name: string]: string } {
+  let variables: { [name: string]: string } = {};
+  const urlSlugs = path.split("/").filter((slug) => slug !== "");
+  if (urlSlugs.length !== querySlugs.length)
+    throw new Error(
+      `Path and query slugs lengths do not match: [${urlSlugs}] AND  [${querySlugs}]`
+    );
+
+  for (let index = 0; index < urlSlugs.length; index++) {
+    // slicing to remove { and } from urlSlugs variable name
+    if (urlSlugs[index].startsWith("{") && urlSlugs[index].endsWith("}")) {
+      const variableName = urlSlugs[index].slice(1, urlSlugs[index].length - 1);
+      variables[variableName] = querySlugs[index];
+    }
+  }
+  return variables;
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> {
+  const queryParams = req.query;
+  const org_id = Number(req.headers["org_id"]);
+  const slug = queryParams.slug as string[];
+  delete queryParams.slug;
+  const method = req.method as RequestMethods;
+
+  const { data: actions, error } = await supabase
+    .from("actions")
+    .select("*")
+    .eq("org_id", org_id)
+    .eq("request_method", method.toLowerCase())
+    .eq("active", true);
+
+  if (error) throw error;
+
+  const matchingAction = org_id
+    ? getMatchingAction(org_id, actions, method, slug)
+    : null;
+
+  const pathParameters =
+    matchingAction?.path?.includes("{") && matchingAction?.path?.includes("}")
+      ? getPathParameters(matchingAction.path, slug)
+      : {};
+
+  const responses = matchingAction?.responses as { [key: string]: any };
+
+  let responseCode;
+  // Search for a 2xx response starting at 200
+  const response =
+    matchingAction && matchingAction.responses
+      ? ((responseCode = Object.entries(matchingAction.responses).find(
+          ([key, value]) => Number(key) >= 200 && Number(key) < 300
+        )?.[0]),
+        responses[responseCode ?? ""] ?? {})
+      : {};
```

Comment:
Imo this code is confusing - I would not use the comma notation thing (hadn't seen it before myself). Are there benefits of the comma approach?

Also, you can use `Object.keys()` rather than `Object.entries()`